lib/puppet/reports/graphite.rb

```diff
@@ -23,11 +26,19 @@ def send_metric payload
 
   def process
     Puppet.debug "Sending status for #{self.host} to Graphite server at #{GRAPHITE_SERVER}"
-    prefix = self.host.split(".").reverse.join(".")
+    
+    if GRAPHITE_REVERSE_HOSTNAME
+      hostname = self.host.split(".").reverse.join(".")      
+    else
+      hostname = self.host
+    end
+    
+    prefix = "#{GRAPHITE_PREFIX}.#{hostname}.#{GRAPHITE_SUFFIX}"
+    
     epochtime = Time.now.utc.to_i
     self.metrics.each { |metric,data|
       data.values.each { |val| 
-        name = "#{prefix}.puppet.#{val[1]}_#{metric}"
+        name = "#{prefix}.#{val[1]}.#{metric}"
```

Comment:
I prefer to have full flexibility about how the metrics are named. People may not want 'puppet' in the metric name (or rather in a different place). More important is replacing the underscore with a dot, so you can split up the metrics. eg. show Success/Failed/Total Events in 1 expression rather than having to define 3 expressions